clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -42,17 +43,32 @@
      * Create a new cluster with the given nodes and partitions
      * @param nodes The nodes in the cluster
      * @param partitions Information about a subset of the topic-partitions this cluster hosts
+     * @deprecated Use the Cluster constructor with 4 parameters
      */
+    @Deprecated
```

Comment:
`Cluster` is a public class (i.e. we publish javadoc for it), so we should not break source or binary compatibility. In practice, it would be surprising if anyone used the constructor outside of tests.